### PR TITLE
Add skipAwait option to Helm v3.Chart

### LINKED ISSUE #1421: Provide a global `kubernetes:skipAwait` setting to turn off all await logic
We should consider a global `kubernetes:skipAwait` setting to turn off all await logic for this package, providing an experience closer to the default Kubernetes API experience.  This has significant downsides (outputs not populated, incorrect deployments not identified, etc.), but for many use cases is the initially "expected" behaviour.

We have support for individual resource-level `skipAwait` annotations today, but not a global setting.

We could also consider support for `skipAwait` on YAML and Helm components.  It's common to want to not await readiness on Helm charts, and is hard to do currently (requires a transformation to add the skipAwait annotation to every child resource).

### PR DESCRIPTION
**Pull request by lblackstone**
### Related issues (optional)
Partial fix for #1421

### CODE REVIEW

**Comment by lblackstone** [comment]
@viveklak @mikhailshilkov I'm still working through .NET and Go, but open to early feedback here

**Comment by viveklak** [comment]
LGTM so far.

File: provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs [hunk 1]
```diff
@@ -25,410 +26,432 @@
...
-            public bool? Verify { get; set; }
-        }
-    }
+		// Set skipAwait annotation on all chart resources.
+		internal ImmutableDictionary<string, object> SkipAwait(ImmutableDictionary<string, object> obj, CustomResourceOptions opts)
+		{
+			var metadata = (ImmutableDictionary<string, object>)obj["metadata"];
+			if (metadata.ContainsKey("annotations"))
+			{
+				metadata = metadata.SetItem("annotations", ((ImmutableDictionary<string, object>)metadata["annotations"]).SetItem("pulumi.com/skipAwait", "true"));
```
Comment (mikhailshilkov) [inline provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs:387]: Nit: consider breaking this line down into several smaller assignments

File: provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs [hunk 2]
```diff
@@ -25,410 +26,432 @@
...
+                obj = obj.SetItem("metadata", metadata);
```
Comment (mikhailshilkov) [inline provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs:388]: Nit: something wrong with indentation on this line. Overall, it looks like you converted this whole file to tabs which isn't common for .NET code. I'd suggest using 4 spaces. (sorry, I hate to comment on tabs vs. spaces)

File: provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs [hunk 3]
```diff
@@ -25,410 +26,432 @@
...
+			}
+			else
+			{
+				obj = obj.SetItem("metadata", metadata.SetItem("annotations", new Dictionary<string, string> { { "pulumi.com/skipAwait", "true" } }.ToImmutableDictionary()));
```
Comment (mikhailshilkov) [inline provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs:392]: Same nit here, maybe a bit too hard to read this as one line

**Review by mikhailshilkov — APPROVED** [review]
LGTM, just some formatting nits

File: provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs (hunk 2 above)
Comment (lblackstone) [reply provider/pkg/gen/dotnet-templates/helm/v3/Chart.cs:388]: Ah, something must have gotten messed up when I copy-pasted to GoLand.

**Pull request merged**